### ISSUE TITLE
Using different passwords when sharing notebook leads to encryption error

### ISSUE THREAD

**Issue by mu88** [laurent22/joplin#6687]
## Environment

Joplin version: `2.7.4`
Platform: Linux
OS specifics: Raspberry Pi 4, arm64/v8, Raspberry Pi OS x64

## Steps to reproduce

1. Create _userA_ with _notebookA_ and encrypt it with `masterPasswordA`.
2. Create _userB_ with _notebookB_ and encrypt it with `masterPasswordB`.
3. Share _notebookA_ between _userA_ and _userB_ and accept the invitation

This fails with the following error:
![image](https://user-images.githubusercontent.com/4560672/180642939-c4c8a228-0ee8-44f5-9ba4-11cf91c9e1dd.png)

As text:
```
Could not respond to the invitation. Please try again, or check with the notebook owner if they are still sharing it.

The error was: "Needs private key <<private key>> to decrypt, but using <<another private key>>"
```

## Describe what you expected to happen
Joplin lets me enter the `masterPasswordA` for _userB_ so that _notebookA_ can be accessed.

**Comment by mu88** [comment]
*push*

**Comment by mu88** [comment]
*push*

**Comment by gitthangbaby** [comment]
Would be nice to solve this by having individual notes content encrypted by different passwords as in apps like Synology/QNap Notes. Instead of confusing E2E (reminds me of Nextcloud attempt) working on unecrypted local storage anyways (and leaking metadata if not [passwords](https://www.reddit.com/r/joplinapp/comments/gg48yv/google_has_your_joplin_encryption_key/)).
Simply open a notebook and ask for a password. If it's a bad password, no luck. That's it. No syncing strategy needs to be taken care of, no master password, no settings at all. An advantage is the simplicity, zero knowledge of password, and access to unencrypted notes on simple systems (terminals) via files.
- Hard to consider having ALL files encrypted, if there is a massive advantage of sync storage in markdown format (=readable) in Joplin versus binaries in the forementioned apps.
- Hard to consider having NO files encrypted, storing some of the sensitive notes in plaintext on the drive despite self hosted is not safe and just can't happen. Plaintext files spread to backups and other destinations very easily. Try Recoll app, index all Samba drives and PC, and search for your password:)
Basically the Notes encryption plugin is the closest solution, if only it was autosaved (not disposable) in encrypted format.

**Comment by mu88** [comment]
ping @laurent22

**Comment by laurent22** [comment]
This has been fixed I believe. You need to update to latest pre-release and latest Joplin Server
